mozilla-release/other-licenses/nsis/Contrib/CliqzHelper/dllmain.cpp

```diff
@@ -0,0 +1,76 @@
+#include <windows.h>
+#include <shlobj.h>
+#include <stdio.h>
+
+#include "installer_tagdata.h"
+#include "helper.h"
+
+// Taken from NSIS
+typedef struct _stack_t {
+  struct _stack_t *next;
+  TCHAR text[1]; // this should be the length of string_size
+} stack_t;
+
+int popstring(stack_t **stacktop, TCHAR *str, int len) {
+  // Removes the element from the top of the stack and puts it in the buffer
+  stack_t *th;
+  if (!stacktop || !*stacktop) {
+    return 1;
+  }
+
+  th = (*stacktop);
+  lstrcpyn(str, th->text, len);
+  *stacktop = th->next;
+  GlobalFree((HGLOBAL)th);
+  return 0;
+}
+
+// Read key value from parameters, try to find this key in
+// tagged area of parent process, if exist - check does
+// distribution file exist. If not - save information about
+// brand into distribution file (in CSIDL_APPDATA\\Cliqz folder)
+extern "C" void __declspec(dllexport) setBrand(HWND hwndParent,
+                                               int string_size,
+                                               TCHAR *variables,
+                                               stack_t **stacktop) {
+#ifdef _DEBUG
+  MessageBox(hwndParent, L"1", L"2", MB_OK);
```

Comment:
Is this still needed? It introduces extra argument `hwndParent`, not used by the main part of code.